### ISSUE TITLE
Error: Failed to install 'MetaboAnalystR' from GitHub

### ISSUE THREAD

**Issue by sarangicodes** [xia-lab/MetaboAnalystR#118]
Hi,

I have the following error when I am trying to install MetaboAnalystR as instructed. 

> devtools::install_github("xia-lab/MetaboAnalystR", build = TRUE, build_opts = c("--no-resave-data", "--no-manual"))
Downloading GitHub repo xia-lab/MetaboAnalystR@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                                 
2: CRAN packages only                  
3: None                                
4: data.table (1.12.6 -> 1.13.0) [CRAN]

Enter one or more numbers, or an empty line to skip updates:3
Installing 49 packages: zoo, sandwich, TH.data, multcomp, plotrix, numDeriv, sn, gbRd, bibtex, SQUAREM, lava, prodlim, timeDate, ipred, gower, checkmate, jpeg, png, estimability, mathjaxr, mutoss, TFisher, Rdpack, pls, pROC, recipes, ModelMetrics, htmlTable, acepack, latticeExtra, Formula, rsm, entropy, metap, spls, igraph, caret, Hmisc, lars, fitdistrplus, pheatmap, ROCR, RJSONIO, som, e1071, randomForest, Cairo, scatterplot3d, Rserve
Installing packages into ‘/Users/msarangi/Library/R/3.6/library’
(as ‘lib’ is unspecified)
Error: Failed to install 'MetaboAnalystR' from GitHub:
  (converted from warning) dependency ‘mnormt’ is not available

I get the same error no matter what number I type, I have update R and everything else is installed as required for my MacOS. Kindly help if you can. I have gone through all the installation error queries and none of the replies have resolved my issue.

Thanks.

**Comment by sarangicodes** [comment]
It is frustrating that I am still unable to install MetaboAnalystR. I have followed all the instructions that have been given on the website, but nothing seems to help. Here are the details:
1. The Website for MetaboAnalystR: https://www.metaboanalyst.ca/MetaboAnalyst/docs/RTutorial.xhtml I am on Mac OS Catalina, Version 10.15.6. 
2. I have R and RStudio. Version R 3.6.3 GUI 1.70 El Capitan build
3. I have XCode installed 
4. I downloaded gFortran-8.2-Mojave and followed the dialogues to install it and it said successfully installed.
5. Regarding GNU Compilers, I have downloaded the mandatory libraries as instructed.

Any help with the error to successfully install this will be highly appreciated. thank you.
```
> library(devtools)
Loading required package: usethis
> devtools::install_github("xia-lab/MetaboAnalystR", build = TRUE, build_vignettes = FALSE)
Downloading GitHub repo xia-lab/MetaboAnalystR@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                                 
2: CRAN packages only                  
3: None                                
4: data.table (1.12.6 -> 1.13.0) [CRAN]

Enter one or more numbers, or an empty line to skip updates:1
plotrix      (NA     -> 3.7-8     ) [CRAN]
numDeriv     (NA     -> 2016.8-1.1) [CRAN]
sn           (NA     -> 1.6-2     ) [CRAN]
gbRd         (NA     -> 0.4-11    ) [CRAN]
bibtex       (NA     -> 0.4.2.2   ) [CRAN]
SQUAREM      (NA     -> 2020.3    ) [CRAN]
lava         (NA     -> 1.6.7     ) [CRAN]
prodlim      (NA     -> 2019.11.13) [CRAN]
timeDate     (NA     -> 3043.102  ) [CRAN]
ipred        (NA     -> 0.9-9     ) [CRAN]
gower        (NA     -> 0.2.2     ) [CRAN]
data.table   (1.12.6 -> 1.13.0    ) [CRAN]
checkmate    (NA     -> 2.0.0     ) [CRAN]
jpeg         (NA     -> 0.1-8.1   ) [CRAN]
png          (NA     -> 0.1-7     ) [CRAN]
estimability (NA     -> 1.3       ) [CRAN]
mathjaxr     (NA     -> 1.0-1     ) [CRAN]
mutoss       (NA     -> 0.1-12    ) [CRAN]
TFisher      (NA     -> 0.2.0     ) [CRAN]
Rdpack       (NA     -> 1.0.0     ) [CRAN]
pls          (NA     -> 2.7-2     ) [CRAN]
pROC         (NA     -> 1.16.2    ) [CRAN]
recipes      (NA     -> 0.1.13    ) [CRAN]
ModelMetrics (NA     -> 1.2.2.2   ) [CRAN]
htmlTable    (NA     -> 2.0.1     ) [CRAN]
acepack      (NA     -> 1.4.1     ) [CRAN]
latticeExtra (NA     -> 0.6-29    ) [CRAN]
Formula      (NA     -> 1.2-3     ) [CRAN]
rsm          (NA     -> 2.10      ) [CRAN]
entropy      (NA     -> 1.2.1     ) [CRAN]
metap        (NA     -> 1.4       ) [CRAN]
spls         (NA     -> 2.2-3     ) [CRAN]
igraph       (NA     -> 1.2.5     ) [CRAN]
caret        (NA     -> 6.0-86    ) [CRAN]
Hmisc        (NA     -> 4.4-0     ) [CRAN]
lars         (NA     -> 1.2       ) [CRAN]
fitdistrplus (NA     -> 1.1-1     ) [CRAN]
pheatmap     (NA     -> 1.0.12    ) [CRAN]
ROCR         (NA     -> 1.0-11    ) [CRAN]
RJSONIO      (NA     -> 1.3-1.4   ) [CRAN]
som          (NA     -> 0.3-5.1   ) [CRAN]
e1071        (NA     -> 1.7-3     ) [CRAN]
randomForest (NA     -> 4.6-14    ) [CRAN]
Cairo        (NA     -> 1.5-12.2  ) [CRAN]
scatterpl... (NA     -> 0.3-41    ) [CRAN]
Skipping 16 packages not available: multtest, mnormt, MSnbase, BiocParallel, siggenes, KEGGgraph, limma, sva, SSPA, genefilter, preprocessCore, Rgraphviz, GlobalAncova, globaltest, pcaMethods, impute
Skipping 1 packages ahead of CRAN: Rserve
Installing 46 packages: plotrix, numDeriv, mnormt, sn, gbRd, bibtex, SQUAREM, lava, prodlim, timeDate, ipred, gower, data.table, checkmate, jpeg, png, estimability, mathjaxr, mutoss, TFisher, Rdpack, pls, pROC, recipes, ModelMetrics, htmlTable, acepack, latticeExtra, Formula, rsm, entropy, metap, spls, igraph, caret, Hmisc, lars, fitdistrplus, pheatmap, ROCR, RJSONIO, som, e1071, randomForest, Cairo, scatterplot3d
Installing packages into ‘/Users/msarangi/Library/R/3.6/library’
(as ‘lib’ is unspecified)
Error: Failed to install 'MetaboAnalystR' from GitHub:
  (converted from warning) package ‘mnormt’ is not available (for R version 3.6.3)
```

**Comment by wyu** [comment]
The problem seems to be related to dependency on R v4 with the current version of mnormt package. I was able to solve it by manually installing an older version instead. 

devtools::install_version('mnormt', version = "1.5-7", repos="http://cran.rstudio.com")
